client/dashboard/src/pages/toolsets/ResourcesTab.tsx

```diff
@@ -0,0 +1,255 @@
+import { Cards } from "@/components/ui/card";
+import {
+  Command,
+  CommandEmpty,
+  CommandGroup,
+  CommandInput,
+  CommandItem,
+  CommandList,
+} from "@/components/ui/command";
+import { MoreActions } from "@/components/ui/more-actions";
+import {
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+} from "@/components/ui/popover";
+import { Type } from "@/components/ui/type";
+import { Resource as GeneratedResource } from "@gram/client/models/components";
+import {
+  useLatestDeployment,
+  useListResources,
+  useUpdateToolsetMutation,
+} from "@gram/client/react-query";
+import { Stack } from "@speakeasy-api/moonshine";
+import { Newspaper } from "lucide-react";
+import { useMemo, useState } from "react";
+import { Toolset, Resource } from "@/lib/toolTypes";
+
+export function ResourcesTabContent({
+  toolset,
+  updateToolsetMutation,
+}: {
+  toolset: Toolset;
+  updateToolsetMutation: ReturnType<typeof useUpdateToolsetMutation>;
+}) {
+  const { data: resourcesResponse } = useListResources({});
+  const allResources = resourcesResponse?.resources ?? [];
+
+  const { data: deployment } = useLatestDeployment(undefined, undefined, {
+    staleTime: 1000 * 60 * 60,
+  });
+
+  // Create a mapping of function ID to function name (slug)
+  const functionIdToName = useMemo(() => {
+    return deployment?.deployment?.functionsAssets?.reduce(
+      (acc, asset) => {
+        acc[asset.id] = asset.name;
+        return acc;
+      },
+      {} as Record<string, string>,
+    );
+  }, [deployment]);
+
+  const toolsetResources = toolset.resources ?? [];
+  const currentResourceUrns = toolset.resourceUrns ?? [];
+
+  const addResourceToToolset = (resourceUrn: string) => {
+    if (currentResourceUrns.includes(resourceUrn)) {
+      return;
+    }
+
+    updateToolsetMutation.mutate({
+      request: {
+        slug: toolset.slug,
+        updateToolsetRequestBody: {
+          resourceUrns: [...currentResourceUrns, resourceUrn],
+        },
+      },
+    });
+  };
+
+  const removeResourceFromToolset = (resourceUrn: string) => {
+    updateToolsetMutation.mutate({
+      request: {
+        slug: toolset.slug,
+        updateToolsetRequestBody: {
+          resourceUrns: currentResourceUrns.filter(
+            (urn) => urn !== resourceUrn,
+          ),
+        },
+      },
+    });
+  };
+
+  // Show empty state if no resources exist at all in the system
+  if (allResources.length === 0) {
+    return (
+      <div className="flex items-center justify-center py-12">
+        <Type muted>No current sources provide resources</Type>
```

Comment:
This is a pretty important state. If you don't want to flesh it out in this PR then just make a ticket